src/main/java/uk/ac/ebi/ddi/xml/validator/utils/Utils.java

```diff
@@ -49,26 +51,32 @@ public static List<Tuple> validateSemantic(Entry entry) {
                             "["+ ERROR+"]" + COLON + REPORT_SPACE + NOT_FOUND_MESSAGE + REPORT_SPACE +
                             DSField.DESCRIPTION));
         }
-
-        if (entry.getDates() != null && !entry.getDates().isEmpty()) {
-            Field.getFields().add(DSField.Date.PUBLICATION);
-            List<Field> fields = Field.getValuesByCategory(FieldCategory.DATE, FieldType.UNKNOWN);
-            for (Field field : fields) {
-                String errorCode = (field.getType() == FieldType.MANDATORY) ? ERROR : WARN;
+        if (entry.getDates() != null) {
+            List<Field> dateFieldList = new ArrayList<Field>();
+            dateFieldList.add(DSField.Date.PUBLICATION);
+            dateFieldList.add(DSField.Date.PUBLICATION_UPDATED);
+            dateFieldList.add(DSField.Date.CREATION);
+            boolean isError = false;
+            if(entry.getDates().isEmpty() || entry.getDates().getDate().isEmpty()){
+                isError = true;
+            } else {
                 boolean found = false;
                 for (Date date : entry.getDates().getDate()) {
-                    if (field.getName().equalsIgnoreCase(date.getType()) && date.getValue() != null
-                            && validateDate(date.getValue())) {
+                    Optional<Field> dateFieldNames = dateFieldList.stream().filter(field -> field.getName().equals(date.getType())).findFirst();
```

Comment:
This might be easier to understand if written as

```
dateFieldList.stream()
  .map(Field::getName)
  .filter(n -> n.equals(date.getType())
  .findFirst()
```